boa/src/syntax/parser/statement/declaration/hoistable/mod.rs

```diff
@@ -89,3 +92,98 @@ where
         }
     }
 }
+
+// This is a helper function to not duplicate code in the individual hoistable deceleration parser functions.
+#[inline]
+#[allow(clippy::too_many_arguments, clippy::type_complexity)]
+fn parse_function_like_declaration<R: Read>(
+    error_context: &'static str,
+    is_default: bool,
+    name_allow_yield: bool,
+    name_allow_await: bool,
+    parameters_allow_yield: bool,
+    parameters_allow_await: bool,
+    body_allow_yield: bool,
+    body_allow_await: bool,
```

Comment:
We could instead create a `CallableDeclaration` trait with all these parameters as implementable getters, then implement it for the four parsers, make `parse_function_like_declaration` accept a `CallableDeclaration` ref and pass that to the function on call. No additional arguments and the definition of the parameters stays in its own module

Reply:
Like this? c1c6438

Reply:
Yep, perfect :)